src/rust/uk.mangainua/src/lib.rs

```diff
@@ -3,13 +3,39 @@
 mod helper;
 
 use aidoku::{
-	error::Result,
+	error::{AidokuError, AidokuErrorKind, Result},
 	prelude::*,
+	std::html::Node,
 	std::net::Request,
 	std::{net::HttpMethod, String, Vec},
 	Chapter, Filter, FilterType, Manga, MangaContentRating, MangaPageResult, MangaStatus,
 	MangaViewer, Page,
 };
+use core::cmp::Ordering;
+
+fn parse_user_hash_and_query(document: Node) -> Result<(String, String)> {
```

Comment:
this doesn't need to consume the document, does it? you shouldn't clone the entire html document if you don't have to
```suggestion
fn parse_user_hash_and_query(document: &Node) -> Result<(String, String)> {
```